file: support ADV files in (File).SegmentFile(..)

adamdecaf opened the pull request:
Issue: https://github.com/moov-io/ach/issues/580

adamdecaf committed:
file: support ADV files in (File).SegmentFile(..)

Issue: https://github.com/moov-io/ach/issues/580

adamdecaf committed:
file: test SegmentFile with Debits

bkmoovio reviewed (approved):
Looks good.  After the merge.  I’ll  go in and modify the C and D for batches.

adamdecaf committed:
file: cleanup SegmentFile for ADV

adamdecaf committed:
file: trim down cyclic complexity in SegmentFile

CI failed due to gocyclo

adamdecaf committed:
file: drop some error checks in SegmentFile

These error checks don't seem possible to hit with valid files (which
are checked at the start) and if SegmentFile generates something
invalid that's a bug.
